### ISSUE TITLE
Strange main window and log behavior on 0.4.9.9h, errors in log console

### ISSUE THREAD

**Issue by ShadowVolt** [mathewthe2/Game2Text#23]
I'm having some strange behavior with version 0.4.9.9h. After some time using the program, the main window appears to get out of sync with the logs, and slowly moves to the more recent lines over time (without any new inputs), instead of staying in sync. At the same time this is occuring, the script-match drop down completely stops working on new lines until the program is rebooted. The lines from before the glitch starts still work though. If I wait until it completely catches up, the drop down starts to work for a short time, then it stops again as new lines come in.

To my untrained eye, it almost looks like it is script-matching at the exact moment the OCR line comes in, even before the main window has a chance to update. Maybe it's matching so fast that the internal logic breaks somehow, forming some kind of loop? Just wildly speculating here, I'm likely quite wrong about that one.

But in any case, every time this happens it throws this error in chrome's console in the log page (specifically the first one, included the second since it may be relevant despite only happening once. I haven't touched the anki part in this install.):

![Capture](https://user-images.githubusercontent.com/17579824/119255488-b95d3200-bb81-11eb-95ee-2691a230bb97.PNG)

---

(Additional info: This is a clean install, aside from the game script. The only non-UI setting I have changed is "match_limit" from the default of 5 down to 2.)

**Comment by mathewthe2** [comment]
I haven't checked yet, but there could a race condition issue with script match updating the main window while the OCR goes in, or it could be a completely different issue.

**Comment by mathewthe2** [comment]
Are you able to recreate this issue with minimal amount of matching?

**Comment by ShadowVolt** [comment]
Yes. I can force it to happen by trying to match very quickly using a game script. I have it on a hotkey with a preset scan region, with that setup if I just scan the same line as fast as I see it pop into the log, it happens within thirty seconds to a minute every time. This is still happening on 0.4.9.9i. It's making it hard to use this program, since every time it happens I have to reboot it.

(Pardon the late reply, have been away for the last couple days.)

**Comment by ShadowVolt** [comment]
Had some free time to poke around at this some more, and it seems that it was introduced in version 0.4.9.9e, since I can't make this bug happen on builds 9d and older. Even mashing the hotkey more than 20 times as fast as I could didn't faze it at all.

Not completely sure where this build lines up here on github, but 9d was the first build that had the log limit and min match feature.
